File: app/src/main/java/com/calintat/explorer/ui/InputDialog.java
package com.calintat.explorer.ui;

import android.app.AlertDialog;
import android.content.Context;
import android.view.View;
import android.widget.EditText;

import com.calintat.explorer.R;

public abstract class InputDialog extends AlertDialog.Builder {

    private final EditText editText;

    protected InputDialog(Context context, String positive, String title) {

        super(context);

        View view = View.inflate(context, R.layout.dialog_edit_text, null);

        editText = (EditText) view.findViewById(R.id.dialog_edit_text);

        setView(view);

        setNegativeButton("Cancel", null);

        setPositiveButton(positive, (dialog, which) -> {

            if (editText.length() != 0) onActionClick(editText.getText().toString());
        });

        setTitle(title);
    }

    public abstract void onActionClick(String text);

    public void setDefault(String text) {

        editText.setText(text);

        editText.setSelection(editText.getText().length());
    }
}

File: app/src/main/java/com/calintat/explorer/ui/InputDialog.kt
package com.calintat.explorer.ui

import android.app.AlertDialog
import android.content.Context
import android.view.View
import android.widget.EditText

import com.calintat.explorer.R

abstract class InputDialog protected constructor(context: Context, positive: String, title: String) : AlertDialog.Builder(context) {

    private val editText: EditText

    init {

        val view = View.inflate(context, R.layout.dialog_edit_text, null)

        editText = view.findViewById(R.id.dialog_edit_text) as EditText

        setView(view)

        setNegativeButton("Cancel", null)

        setPositiveButton(positive) { _, _ ->

            if (editText.length() != 0) onActionClick(editText.text.toString())
        }

        setTitle(title)
    }

    abstract fun onActionClick(text: String)

    fun setDefault(text: String) {

        editText.setText(text)

        editText.setSelection(editText.text.length)
    }
}